lib/prometheus.js

```diff
@@ -0,0 +1,135 @@
+const PromClient = require('prom-client');
+const uuid = require("uuid");
+const debug = require('debug')('plugin:publish-metrics:prometheus');
+
+const COUNTERS_STATS = 'counters', // counters stats
+  RATES_STATES = 'rates', // rates stats
+  SUMMARIES_STATES = 'summaries'; // summaries stats includes errors
+
+class PrometheusReporter {
+
+  constructor(config, events) {
+    this.jobUUID = uuid.v4();
```

Comment:
Would it make sense to make this configurable? (to e.g. be able to set it in tests ran by the Operator to match the job ID in K8S)

Reply:
`this.jobUUID` is analogous to the Pod ID rather than a Job ID in K8s. Prometheus refers to "workers" as jobs.

But, I see what that might be a good idea.

For our use case, with the order of how things get run, configuring the `this.jobUUID` with the Pod ID is not do-able. That is, it's too late.

Regardless, we can use an env var, e.g. `WORKER_ID`,  and fallback `uuid.v4()` if non is provided.

Reply:
Done! 

Operator Jobs will use the [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api) to expose the pod's `hostname` (also used to name the pod) as the `WORKER_ID`.